lib/internal/url.js

```diff
@@ -1377,8 +1377,14 @@ function getPathFromURL(path) {
   return isWindows ? getPathFromURLWin32(path) : getPathFromURLPosix(path);
 }
 
+// We percent-encode % character when converting from file path to URL,
+// as this is the only character that won't be percent encoded by
+// default URL percent encoding when pathname is set.
+const percentRegEx = /%/g;
 function getURLFromFilePath(filepath) {
   const tmp = new URL('file://');
+  if (filepath.indexOf('%') !== -1)
```

Comment:
@refack also asked it but why not `filepath.includes('%')`?

Reply:
No reason other than habit, can change.